tasks/vg-default.yml

```diff
@@ -4,12 +4,19 @@
 #
 # TODO: Decide on semantics. Playbook PV specs should probably be complete to enable idempotency.
 #
-- name: Install LVM2 commmands as needed
-  package:
-    name: lvm2
-    state: present
-  when: pool.type == "lvm"
-  
+- block:
+  - name: Install LVM2 commands as needed
+    package:
+      name: lvm2
+      state: present
+  - set_fact:
+      storage_packages: "{{ storage_packages + ['lvm2'] }}"
+  when: "'lvm2' not in storage_packages and pool.type == 'lvm' and not ansible_check_mode"
+
+- name: Update ansible_facts after installing lvm2
+  setup:
+  run_once: true
```

Comment:
Can you move the `setup` call into the `block` above?

Reply:
I tried that in the initial development stage. Unfortunately, I get the following error when I try to move `setup` into the task block above: 

```
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'pool.type == \"lvm\" and pool.name in ansible_facts.lvm.vgs' failed. The error was: error while evaluating conditional (pool.type == \"lvm\" and pool.name in ansible_facts.lvm.vgs): 'dict object' has no attribute 'lvm'\n\nThe error appears to have been in '/home/tim/Documents/storage/tasks/vg-default.yml': line 43, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n- block:\n    - name: collect list of current pvs\n      ^ here\n"}
```